services/app-api/src/resolvers/contract/contractResolver.ts

```diff
@@ -99,5 +107,103 @@ export function contractResolver(): Resolvers['Contract'] {
+            // add a span to OTEL
+            const span = tracer?.startSpan(
+                'fetchContractWithQuestionsResolver',
+                {},
+                ctx
+            )
+            setResolverDetailsOnActiveSpan(
+                'fetchContractWithQuestions',
+                user,
+                span
+            )
+
+            const questionsForContract = await store.findAllQuestionsByContract(
+                parent.id
+            )
+
+            if (questionsForContract instanceof Error) {
+                const errMessage = `Issue finding contract message: ${questionsForContract.message}`
+                setErrorAttributesOnActiveSpan(errMessage, span)
+
+                if (questionsForContract instanceof NotFoundError) {
+                    throw new GraphQLError(errMessage, {
+                        extensions: {
+                            code: 'NOT_FOUND',
+                            cause: 'DB_ERROR',
+                        },
+                    })
+                }
+
+                throw new GraphQLError(errMessage, {
+                    extensions: {
+                        code: 'INTERNAL_SERVER_ERROR',
+                        cause: 'DB_ERROR',
+                    },
+                })
+            }
+
+            const dmcoQuestions = questionsForContract
+                .filter((question) => question.division === 'DMCO')
+                .map((question) => {
+                    return {
+                        node: {
+                            ...question,
+                        },
+                    }
+                })
+            const dmcpQuestions = questionsForContract
+                .filter((question) => question.division === 'DMCP')
+                .map((question) => {
+                    return {
+                        node: {
+                            ...question,
+                        },
+                    }
+                })
+            const oactQuestions = questionsForContract
+                .filter((question) => question.division === 'OACT')
+                .map((question) => {
+                    return {
+                        node: {
+                            ...question,
+                        },
+                    }
+                })
+
+            const questionPayload: IndexQuestionsPayload = {
+                DMCOQuestions: {
+                    totalCount: dmcoQuestions.length,
+                    edges: dmcoQuestions,
+                },
+                DMCPQuestions: {
+                    totalCount: dmcpQuestions.length,
+                    edges: dmcpQuestions,
+                },
+                OACTQuestions: {
+                    totalCount: oactQuestions.length,
+                    edges: oactQuestions,
+                },
+            }
+            // A state user cannot access contracts that don't belong to their state
+            if (isStateUser(user)) {
```

Comment:
this should be impossible, right? We're already checking for this in fetchContract so I'm not sure how we could get to the contract resolver without having checked this.

In my mind, authorization should happen at the root resolver layer and hopefully we can skip it in the leaf resolvers.